### PR TITLE
feat: add snooze

### LINKED ISSUE #10968: Mail: Snooze messages
For https://github.com/nextcloud/mail/issues/2048

### PR DESCRIPTION
**Pull request by JohannesGGE**
### ☑️ Resolves

* Fix #10968 

### 🖼️ Screenshots

user_manuel
image

admin_manuel
image

### CODE REVIEW

File: user_manual/groupware/mail.rst [hunk 1]
```diff
@@ -156,3 +156,16 @@ Unsubscribe from a mailing list
...
+
+.. versionadded:: 3.4
+
+Snoozing a message or thread moves it in a dedicated mailbox until the selected snooze date is reached and the message or thread is moved back to the original mailbox.
+
+1. Open action menu of an envelope or thread
+2. Click *Snooze*
+3. Select how long the message or thread should be snoozed
+
+.. note:: If AJAX is selected for cron job execution in the admin settings, the snooze feature is deactivated.
```
Comment (ChristophWurst) [inline user_manual/groupware/mail.rst:171]: not sure if we want to give this information to users. we could move this information into the mail admin docs

Comment (JohannesGGE) [reply user_manual/groupware/mail.rst:171]: :+1: Are there any other features that are disabled with ajax (beside snooze and scheduled sending) worth mentioning?

Comment (ChristophWurst) [reply user_manual/groupware/mail.rst:171]: those should be the only two

File: user_manual/groupware/mail.rst [hunk 2]
```diff
@@ -156,3 +156,16 @@ Unsubscribe from a mailing list
 .. versionadded:: 3.1
 
 Some mailing lists and newsletters allow to be unsubscribed easily. If the Mail app detects messages from such a sender, it will show an *Unsubscribe* button next to the sender information. Click and confirm to unsubscribe from the list.
+
+Snooze
+~~~~~~
+
+.. versionadded:: 3.4
+
+Snoozing a message or thread moves it in a dedicated mailbox until the selected snooze date is reached and the message or thread is moved back to the original mailbox.
```
[the remark this answers is not in the document]
Comment (JohannesGGE) [reply user_manual/groupware/mail.rst]: :white_check_mark:

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**